quandl/model/merged_dataset.py

```diff
@@ -141,30 +158,33 @@ def _initialize_raw_data(self, datasets):
             self._raw_data[k] = getattr(self, k)
         return self._raw_data
 
-    def _get_dataset(self, dataset_code, **options):
+    def _build_dataset_object(self, dataset_code, **options):
         options_copy = options.copy()
-        # data_codes are tuples e.g., ('GOOG/NASDAQ_AAPL', {'column_index":
-        # [1,2]}) or strings 'NSE/OIL'
+        # data_codes are tuples
+        # e.g., ('GOOG/NASDAQ_AAPL', {'column_index": [1,2]})
+        # or strings
+        # e.g., 'NSE/OIL'
         code = self._get_request_dataset_code(dataset_code)
 
         dataset = Dataset(code, None, **options_copy)
-        dataset.column_index = []
+        # save column_index param requested dynamically
+        # used later on to determine:
+        # if column_index is an array, fetch all data and use locally to filter columns
+        # if column_index is an empty array, fetch all data and don't filter columns
+        dataset.requested_column_indexes = self._get_req_dataset_col_indexes(dataset_code, code)
+        return dataset
+
+    def _get_req_dataset_col_indexes(self, dataset_code, code_str):
         # ensure if column_index dict is specified, value is a list
         params = self._get_request_params(dataset_code)
         if 'column_index' in params:
             column_index = params['column_index']
             if not isinstance(column_index, list):
                 raise ValueError(
-                    "%s : column_index needs to be a list of integer indexes" % code)
-            dataset.column_index = column_index
-        # if column_indexes are specified, check that they actually exist
-        max_column_index = len(dataset.column_names) - 1
-        for index in dataset.column_index:
-            if index > max_column_index or index < 1:
-                raise ValueError("%s : requested index %s is out of range. \
-                                  Min index is 1 and Max index is %s" % (code, index,
-                                                                         max_column_index))
-        return dataset
+                    "%s : column_index needs to be a list of integer indexes" % code_str)
+            return column_index
```

Comment:
Does this return a number or a list of numbers. If it's a single number it's a bit odd the default response is a list. If its a list no problem.